Factory/OidcTokenResponseFactory.php

```diff
@@ -85,11 +86,22 @@ public function createOidcTokenResponse(array $responseData, OidcGrantInterface
   }
 
   /**
-   * @param string[] $mandatoryClaims
+   * @param string[]|array<string, array> $mandatoryClaims
    *
    * @return self
    */
   public function setMandatoryClaims(array $mandatoryClaims): self {
+    if (\array_is_list($mandatoryClaims)) {
```

Comment:
FYI: If the parameter can be sequential without being in order, the array_is_list function won't cut it.

Reply:
The check is there to keep backwards compatibility, where we could do `setMandatoryClaims(['some_claim'])`, now we can set mandatory claims per token type.

If someone sent a list of mandatory claims that is was not a list previously, it will break, but that was not the intention, so I think this is fine.